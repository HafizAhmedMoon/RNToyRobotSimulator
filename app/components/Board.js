import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, View } from "react-native";

export class Board extends React.Component {

  static propTypes = {
    rows: PropTypes.number.isRequired,
    cols: PropTypes.number.isRequired,
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
  };

  render() {
    const {rows, cols, x, y} = this.props;
    const matrix = Array.from(new Array(rows), (val, row) => {
      const elms = Array.from(new Array(cols), (val, col) => {
        return <View key={col} style={[styles.col, row === x && col === y && styles.selected]}/>
      });
      return <View key={row} style={styles.row}>{elms}</View>
    }).reverse();
    return (
      <View style={styles.container}>
        {matrix}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderColor: '#aaa',
    borderWidth: 1,
  },
  row: {
    flexDirection: 'row'
  },
  col: {
    borderColor: '#aaa',
    borderWidth: 1,
    width: 50,
    height: 50,
  },
  selected: {
    backgroundColor: 'blue'
  },

});
