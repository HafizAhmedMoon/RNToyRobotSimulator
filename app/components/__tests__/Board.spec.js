import React from 'react';
import { shallow } from 'enzyme';
import { Board } from "../Board";

describe("Board Component", () => {

  it('renders as expected', () => {
    const wrapper = shallow(
      <Board rows={5} cols={5} x={2} y={4}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
