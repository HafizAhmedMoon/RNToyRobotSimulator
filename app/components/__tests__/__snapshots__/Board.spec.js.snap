// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component renders as expected 1`] = `
<View
  style={
    Object {
      "borderColor": "#aaa",
      "borderWidth": 1,
    }
  }
>
  <View
    key="4"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="1"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="2"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="3"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="4"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
  </View>
  <View
    key="3"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="1"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="2"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="3"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="4"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
  </View>
  <View
    key="2"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="1"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="2"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="3"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="4"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          Object {
            "backgroundColor": "blue",
          },
        ]
      }
    />
  </View>
  <View
    key="1"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="1"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="2"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="3"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="4"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
  </View>
  <View
    key="0"
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      key="0"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="1"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="2"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="3"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
    <View
      key="4"
      style={
        Array [
          Object {
            "borderColor": "#aaa",
            "borderWidth": 1,
            "height": 50,
            "width": 50,
          },
          false,
        ]
      }
    />
  </View>
</View>
`;
